src/functions/swagger/functions.yaml

```diff
@@ -247,132 +88,50 @@ parameters:
-    description: |
-      An object specifying a library.
-
-      [NOTE]
-      ====
-      The `name` property in the library object must match the library name specified in the path.
-      ====
-    schema:
-      $ref: "#/definitions/Libraries"
-
-  BodyFunction:
-    in: body
-    name: Function Definition
+    name: functions
     required: true
-    description: |
-      An object specifying a function.
-
-      [NOTE]
-      ====
-      The `name` property in the function object must match the function name specified in the path.
-      ====
+    description: The JavaScript code for all functions in the library.
     schema:
-      $ref: "#/definitions/Functions"
+      type: string
+      example: function f2c(f) { return (5/9)*(f-32); }
 
 definitions:
   Libraries:
     type: object
     title: Library Definition
     properties:
-      name:
-        type: string
-        description: The name of the library.
-        example: math
-      functions:
-        type: array
-        description: An array of objects, each of which gives information about one function.
-#       example:
-#         - name: adder
-#           id: d8:91:45:37:79:75:eb:15
-#           code: function adder(a, b) { return a + b; }
-#         - name: multiplier
-#           id: 8e:ab:89:e6:1e:c2:21:1a
-#           code: function multiplier(a, b) { return a * b; }
-        items:
-          $ref: '#/definitions/Functions'
-    required:
-      - name
-      - functions
-
-  Functions:
-    type: object
-    title: Function Definition
-    properties:
-      name:
-        type: string
-        description: |
-          The name of the function.
-          This must match the name of the JavaScript function that returns the result, as specified by the `code` property.
-        example: f2c
-      id:
-        type: string
-        description: |
-          A unique ID for the function.
-          You do not need to specify an ID when creating or updating a function.
-          An ID is generated automatically if you do not specify one.
-        example: c4:1a:39:46:46:88:ff:3d
-      code:
+      __library__:
         type: string
         description: |
-          The JavaScript code for the function.
-          The name of the JavaScript function that returns the result must match the name specified by the `name` property.
-          The JavaScript code may contain other helper functions.
-        example: function f2c(f) { return (5/9)*(f-32); }
+          The JavaScript code for all functions in the library.
+
+          The name of the property is the name of the library.
+        example: function add(a, b) { return a + b; }
+                 function mul(a, b) { return a * b; }
     required:
-      - name
-      - code
+      - __library__
 
 responses:
   OK:
     description: The operation was successful.
+    schema:
+      type: string
 
   LibraryNotFound:
     description: |
       Not found.
       The library name in the path may be incorrect.
```

Comment:
For GET, POST, DELETE under all REST endpoints it should be worth adding HTTP 406  "Status Not Acceptable" if user doesn't conform to the schema. For example, if I give "evaluator/v1/libraries/math/science" evaluator returns HTTP 406.

Reply:
Done